### PR TITLE
Documentation MCU

### PR DESCRIPTION
**Pull request by jmuellerstudentacc**
Besides the code documentation I have also reconfigured the code a little bit. The encoders are now in the HAL level and all interrupts + the timer from the MCU is called in the main file. It is also tested on the robot if it is working as expected.

### CODE REVIEW

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
 #define ADAS_DEBUG
 
 //PLS Buffer
+/** Size of PLS receiving buffer in bits */
 #define PLS_RCV_BUFF_SIZE                                   500U
+/** Size of PLS sending buffer in bits */
 #define PLS_SND_BUFF_SIZE                                   50U
+
 //ICC Buffer
+/** Size of ICC receiving buffer in bits */
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Typo from my side.
```suggestion
/** Size of ICC receiving buffer in bytes*/
```

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
 #define ICC_RCV_BUFF_SIZE                                   ICC_LEN
+/** Size of ICC sending buffer in bits */
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Typo from my side.
```suggestion
/** Size of ICC sending buffer in bytes*/
```

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 3]
```diff
@@ -39,45 +53,89 @@
...
+/** Direction for Right Motor*/
 #define PIN_DIRECTION_R                                 48
+/** Encoder Interrupt for Right Motor*/
 #define PIN_ENC_R                                        2
+/** Encoder Interrupt for Left Motor*/
 #define PIN_ENC_L                                        3
+/** External Start Button*/
 #define Btn_Start                                       46
+/** Voltage Level of Battery*/
 #define PIN_VBAT                                        A0
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: You don't need this line on your core. Voltage will be measured via NavU.

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 4]
```diff
@@ -39,45 +53,89 @@
 #define SICK_NAK                                        0x15
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Not needed on the MCU.

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 5]
```diff
@@ -39,45 +53,89 @@
...
 #define PIN_ENC_R                                        2
+/** Encoder Interrupt for Left Motor*/
 #define PIN_ENC_L                                        3
+/** External Start Button*/
 #define Btn_Start                                       46
+/** Voltage Level of Battery*/
 #define PIN_VBAT                                        A0
 
 
 //Navigation parameters
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Navigation parameters not needed on the MCU

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h [hunk 6]
```diff
@@ -39,45 +53,89 @@
...
+/** Inter-Controller Communication: Command to continue movement */
 #define ICC_CMD_CONT_DRIVE                              0x08
+/** Inter-Controller Communication: Command to acknowledge message */
 #define ICC_CMD_FB_ACK                                  0x01
+/** Inter-Controller Communication: Command to acknowledge reached distance */
 #define ICC_CMD_FB_DIST                                 0x02
+/** Inter-Controller Communication: Command to acknowledge reached rotation */
 #define ICC_CMD_FB_ROT                                  0x03
 
 //Environmetal parameters
```
Comment (christophjurczyk) [inline 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Environmetal parameters not needed on MCU.

Comment (jmuellerstudentacc) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: We are sending this to you as a feedback

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h (hunk 3 above)
Comment (jmuellerstudentacc) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: removed

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h (hunk 4 above)
Comment (jmuellerstudentacc) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: removed

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h (hunk 5 above)
Comment (jmuellerstudentacc) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: removed

File: 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h (hunk 6 above)
Comment (jmuellerstudentacc) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Ahh.. I see you have marked the wrong line so I was confused :-D removed it

Comment (christophjurczyk) [reply 06_source/ADAS/ADAS_MCU/ADAS_Cfg.h]: Yeah...sorry my mistake.

**Pull request merged**